src/bootstrap/dist.rs

```diff
@@ -98,6 +98,137 @@ pub fn docs(build: &Build, stage: u32, host: &str) {
+            path .iter()
+                .map(|dir| dir.join(file))
+                .find(|p| fs::metadata(&p).is_ok());
+
+        if let Some(file_path) = file_path {
+            found.push(file_path);
+        } else {
+            panic!("Could not find '{}' in {:?}", file, path);
+        }
+    }
+
+    found
+}
+
+fn make_win_dist(rust_root: &Path, plat_root: &Path, target_triple: &str) {
+    //Ask gcc where it keeps its stuff
+    let gcc_out =
+        String::from_utf8(
+            Command::new("gcc.exe")
+                .arg("-print-search-dirs")
+                .output()
+                .expect("failed to execute gcc")
+                .stdout).expect("gcc.exe output was not utf8");
+    let mut bin_path: Vec<_> =
+        env::split_paths(&env::var_os("PATH").expect("failed to get $PATH"))
+        .collect();
+    let mut lib_path = Vec::new();
+
+    for line in gcc_out.lines() {
+        let idx = line.find(':').unwrap();
+        let key = &line[..idx];
+        let trim_chars: &[_] = &[' ', '='];
+        let mut value: Vec<_> =
+            line[(idx + 1)..]
+                .trim_left_matches(trim_chars)
+                .split(';')
+                .map(|s| PathBuf::from(s))
+                .collect();
+
+        if key == "programs" {
+            bin_path.append(&mut value);
+        } else if key == "libraries" {
+            lib_path.append(&mut value);
+        }
+    }
+
+    let target_tools = vec!["gcc.exe", "ld.exe", "ar.exe", "dlltool.exe"];
+    let mut rustc_dlls = vec!["libstdc++-6.dll", "libwinpthread-1.dll", "libwinpthread-1.dll"];
+    if target_triple.starts_with("i686-") {
+        rustc_dlls.push("libgcc_s_dw2-1.dll");
+    } else {
+        rustc_dlls.push("libgcc_s_seh-1.dll");
+    }
+
+    let target_libs = vec![ //MinGW libs
+        "libgcc.a",
+        "libgcc_eh.a",
+        "libgcc_s.a",
+        "libm.a",
+        "libmingw32.a",
+        "libmingwex.a",
+        "libstdc++.a",
+        "libiconv.a",
+        "libmoldname.a",
+        "libpthread.a",
+        //Windows import libs
+        "libadvapi32.a",
+        "libbcrypt.a",
+        "libcomctl32.a",
+        "libcomdlg32.a",
+        "libcrypt32.a",
+        "libgdi32.a",
+        "libimagehlp.a",
+        "libiphlpapi.a",
+        "libkernel32.a",
+        "libmsvcrt.a",
+        "libodbc32.a",
+        "libole32.a",
+        "liboleaut32.a",
+        "libopengl32.a",
+        "libpsapi.a",
+        "librpcrt4.a",
+        "libsetupapi.a",
+        "libshell32.a",
+        "libuser32.a",
+        "libuserenv.a",
+        "libuuid.a",
+        "libwinhttp.a",
+        "libwinmm.a",
+        "libwinspool.a",
+        "libws2_32.a",
+        "libwsock32.a",
+    ];
+
+    //Find mingw artifacts we want to bundle
+    let target_tools = find_files(target_tools, &bin_path);
+    let rustc_dlls = find_files(rustc_dlls, &bin_path);
+    let target_libs = find_files(target_libs, &lib_path);
+
+    fn copy(src: &Path, dest_folder: &Path) {
```

Comment:
It looks to me like that function assumes `dst` already refers to a full path name like "C:\rust\test.dll" instead of just the folder name which is what this function handles. I can change this function to call `util::copy` instead of `fs::copy` though.